runtime/lua/vim/uri.lua

```diff
@@ -83,6 +83,14 @@ local WINDOWS_URI_SCHEME_PATTERN = '^([a-zA-Z]+[a-zA-Z0-9.+-]*):[a-zA-Z]:.*'
 ---@param bufnr number
 ---@return string URI
 local function uri_from_bufnr(bufnr)
+  local relative_name
+  vim.api.nvim_buf_call(bufnr, function()
+    relative_name = vim.api.nvim_exec(":echo @%", true)
+  end)
```

Comment:
```suggestion
  local relative_name = vim.fn.bufname(bufnr)
```
Reusing `fname` doesn't work here?

Reply:
If this is supposed to be relative to the project directory, that's not going to be a robust way of doing it.

Reply:
nice your suggestion works

Reply:
Still though, is this supposed to be a path relative to the project directory? Because path returned from `@%` and `bufname()` is relative to the current working directory. Sometimes. And sometimes it's a full path.

Reply:
Frankly, I don't fully understand it. Are those `deno:` buffers created by the language server? If they don't actually exist on the filesystem then the buffers should use a custom `deno://` scheme. Then `nvim_buf_get_name` will leave them as they are.

Reply:
The document is a deno virtual text (more info here <https://deno.land/manual/language_server/overview>) so it doesn't exist on the disk at all.

All I know Is that `bufname` returns the correct path which is for example: `deno:/https/deno.land/x/simple_shell%400.13.0/mod.ts` instead of the current broken path `"~/dev/deno/lab/qsd/deno:/https/deno.land/x/simple_shell%400.13.0/mod.ts" `

My point is with custom schema, the implementer probably don't care about the path at all, they just wants the url since they're the one that knows how to use it.

Reply:
Unfortunately it seems like vim only "recognizes" URI schemes that contain `://`. Can `deno:` be translated into `deno://` when those buffers are created?

Reply:
I already tried that but deno lsp don't accept `deno://` (with double slash)
It only works with `deno:/` (one slash)

Reply:
Yes, it would have to be translated on the way in and out.

Reply:
Another question, when there are multiple denols instances running, can they open the same `deno:` file twice, but with different contents? If yes then it might be necessarily to also have a language server identifier in the path, because you can't have duplicate file names. (like `deno://1/path/to/something`, with `1` being the identifier)

Reply:
I'm not even sure how to have multiple denols instances? (is there a use case?) 
I think `deno:path` should always point to the same file anyway because its readonly. (virtual text is meant only for browsing code not modifying it)

Reply:
Where does vim recognize schema btw, it seems vim is in the wrong here
As far as I understood from the spec https://url.spec.whatwg.org/#scheme-state seeing `:` is enough to consider the name before it a schema

Reply:
Multiple denols instances as in multiple deno projects opened in the same neovim instance.

Yes, `:` without `//` is technically correct, but it's just not handled in vim. I'm not sure where it is, you'd have to search for it in the C part of the code base.